product_visible_discount_extension/models/sale.py

```diff
@@ -0,0 +1,189 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright (C) 2015 Savoir-faire Linux
+#    (<http://www.savoirfairelinux.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.exceptions import ValidationError
+from openerp import (
+    models,
+    fields,
+    api,
+)
+
+
+class SaleOrderLine(models.Model):
+
+    """
+    Compute the value of visible_discount and price_unit.
+
+    As those two fields are mutual, we have to store a value in a
+    temp variable.
+
+    Any change to visible_discount will trigger onchanges events in that order:
+        _onchange_visible_discount
+        _onchange_price
+
+    Any change to price_unit will trigger onchange events in that order:
+        _onchange_price
+        _onchange_visible_discount
+    """
+
+    _name = 'sale.order.line'
+    _inherit = 'sale.order.line'
+
+    list_price = fields.Float('Price list',
+                              related="product_id.list_price",
+                              store=True,
+                              readonly=True)
+    visible_discount = fields.Float('Customer  Discount (%)')
```

Comment:
There is a problem, if I use the real discount, it modify the `price_unit` value to the actual price it would get with the discount calculated.. Which means we can't calculate the discount again. If we were using the discount field, we would have something like this:

    price_unit = (list_price * (1 - discount))
    discount = discount
    total price = price_unit * (1 - discount)

The result would be to calculate twice the same discount. The problem comes from openerp not showing the price after discount.

One other way to do, would be to use the price_unit as list_price without touching it. Calculating the discount and having an other field to calculate the price after discount. In the end, the discount field is used as usual, the price_unit would be exactly the same as list_price. The price after discount would have no effect anywhere else. 

But we would want to show the price after discount in reports.